src/grt/src/GlobalRouter.cpp

```diff
@@ -2286,7 +2346,52 @@ GSegment GlobalRouter::createFakePin(Pin pin,
   pin_connection.final_x = std::max(x_tmp, pin_connection.final_x);
   pin_connection.final_y = std::max(y_tmp, pin_connection.final_y);
 
-  return pin_connection;
+  pad_pins_connections_[net->getDbNet()].push_back(pin_connection);
+
+  auto nets_pins = net->getPins();
+  int tile_size = grid_->getTileSize();
+
+  if (pin_connection.init_y == pin_connection.final_y) {
+    int die_area_min_x = grid_->getXMin();
+
+    int init_id_x
+        = floor((float) ((pin_connection.init_x - die_area_min_x) / tile_size));
+    int final_id_x = floor(
+        (float) ((pin_connection.final_x - die_area_min_x) / tile_size));
+
+    for (Pin& net_pin : nets_pins) {
+      if (!(net_pin.getITerm() == pin.getITerm())) {
+        auto net_pin_pos = net_pin.getOnGridPosition();
+        int net_pin_id_x
+            = floor((float) ((net_pin_pos.x() - die_area_min_x) / tile_size));
+        if ((net_pin_id_x >= init_id_x) && (net_pin_id_x <= final_id_x)) {
+          pin_position.setX(original_x);
+          pad_pins_connections_[net->getDbNet()].pop_back();
+          break;
```

Comment:
Following my first comment about adding the `pin_connection` to the map, you can remove the `pop_back` line and use `return` instead of break. This way, you will still update the `pin_position`, but will not creat a connection when an instance pin is inside the pin offset.